netbox/dcim/models/devices.py

```diff
@@ -983,6 +983,13 @@ def clean(self):
                 'vc_position': _("A device assigned to a virtual chassis must have its position defined.")
             })
 
+        if hasattr(self, 'vc_master_for') and self.vc_master_for and self.vc_master_for != self.virtual_chassis:
+            raise ValidationError({
+                'virtual_chassis': _("Cannot change to a different virtual chassis when assigned as a master. Remove it as the master of {master} first.").format(
```

Comment:
Can we tweak the error message to avoid confusing the user? (We're not necessarily moving the device to a _different_ virtual chassis; we may just be removing it.) Something like "Device cannot be removed from virtual chassis {virtual_chassis} because it is currently designated as its master."